### ISSUE TITLE
Compound LoginIDs as first class citizen

### ISSUE THREAD

**Issue by kaojohnny** [SkygearIO/features#296]
# Description

In issue https://github.com/SkygearIO/features/issues/294, we suggest user to use JSON when need login/signup by compound loginIDs.

```
signupWithLoginIDs({
  email: 'example@example.com',
  role_phone: '[\"admin\",\"phone\"]' // JSON string
}, 'password')
```

In this feature spec, compound LoginIDs should be treated as the first class citizen. 

```
signupWithLoginIDs({
  email: 'example@example.com',
  role_phone: {
    role: 'admin',
    phone: '1234567'
  }
}, 'password')
```

We could support JSON or other formation at the API endpoint, and have the appropriate interface on each client SDK.

The end goal should be, people should feel comfortable to use LoginIDs to store the data, so they don't need to "sync" the data in loginIDs and other places (such as metadata)

And also maybe if we consider this feature seriously, we want to handle the case of "login by Role + Username or Role + Email" too.

**Comment by tensiuyan** [comment]
from Ben:

Specification wise, can we suggest that maybe another simpler approach for compound key LoginID, is we can support string as login ID only, instead we add a few called something like domain:String, to suppose the case of "Role" + "LoginID"? It seems more straight forward and easy. Compound key could be too complicated and don't have much use case beyond a simple "Domain" + "LoginID" model.

**Comment by chpapa** [comment]
Override and fixed by #325
